python/src/main/java/org/apache/zeppelin/python/PythonInterpreter.java

```diff
@@ -0,0 +1,261 @@
+        gatewayServer = new GatewayServer(this, port);
+        gatewayServer.start();
+        logger.info("Bootstrap inputs with " + BOOTSTRAP_INPUT_PY);
+        bootStrapInterpreter(BOOTSTRAP_INPUT_PY);
+      } catch (IOException e) {
+        logger.error("Can't execute " + BOOTSTRAP_INPUT_PY + " to " +
+            "initialize Zeppelin inputs in python process", e);
+      }
+    }
+
+
+  }
+
+  @Override
+  public void close() {
+
+    logger.info("closing Python interpreter .....");
+    try {
+      process.close();
+      gatewayServer.shutdown();
+    } catch (IOException e) {
+      logger.error("Can't close the interpreter", e);
+    }
+
+  }
+
+
+  @Override
+  public InterpreterResult interpret(String cmd, InterpreterContext contextInterpreter) {
+
+    this.context = contextInterpreter;
+
+    String output = sendCommandToPython(cmd);
+    return new InterpreterResult(Code.SUCCESS, output.replaceAll(">>>", "")
+        .replaceAll("\\.\\.\\.", "").trim());
+  }
+
+
+  @Override
+  public void cancel(InterpreterContext context) {
+    try {
+      process.interrupt();
+    } catch (IOException e) {
+      logger.error("Can't interrupt the python interpreter", e);
+    }
+  }
+
+  @Override
+  public FormType getFormType() {
+    return FormType.NATIVE;
+  }
+
+  @Override
+  public int getProgress(InterpreterContext context) {
+    return 0;
+  }
+
+  @Override
+  public Scheduler getScheduler() {
+    return SchedulerFactory.singleton().createOrGetParallelScheduler(
+        PythonInterpreter.class.getName() + this.hashCode(), 10);
+  }
+
+  @Override
+  public List<String> completion(String buf, int cursor) {
+    return null;
+  }
+
+  public PythonProcess getPythonProcess() {
+    if (process == null)
+      return new PythonProcess(getProperty(ZEPPELIN_PYTHON));
+    else
+      return process;
+  }
+  
+  private Job getRunningJob(String paragraphId) {
+    Job foundJob = null;
+    Collection<Job> jobsRunning = getScheduler().getJobsRunning();
+    for (Job job : jobsRunning) {
+      if (job.getId().equals(paragraphId)) {
+        foundJob = job;
+      }
+    }
+    return foundJob;
+  }
+
+
+  private String sendCommandToPython(String cmd) {
+
+    String output = "";
+    logger.info("Sending : \n " + cmd);
+    try {
+      output = process.sendAndGetResult(cmd);
+    } catch (IOException e) {
+      logger.error("Error when sending commands to python process", e);
+    }
+
+    return output;
+  }
+
+
+  private void bootStrapInterpreter(String file) throws IOException {
+
+    BufferedReader bootstrapReader = new BufferedReader(
+        new InputStreamReader(
+            PythonInterpreter.class.getResourceAsStream(file)));
+    String line = null;
+    String bootstrapCode = "";
+    while ((line = bootstrapReader.readLine()) != null) {
+      bootstrapCode += line + "\n";
+    }
+
+    if (py4J && port != null && port != -1)
+      bootstrapCode = bootstrapCode.replaceAll("\\%PORT\\%", port.toString());
```

Comment:
This need to be updated to conform the [project's code conventions](https://github.com/apache/incubator-zeppelin/blob/master/CONTRIBUTING.md#code-convention)